lib/Optimizer/Transforms/Mapping.cpp

```diff
@@ -671,11 +680,19 @@ struct Mapper : public cudaq::opt::impl::MappingPassBase<Mapper> {
     // and stop once you hit a used one. If you removed from the middle, you
     // would renumber the qubits, which would invalidate the mapping indices.
     for (unsigned i = sources.size() - 1; i >= numOrigQubits; i--) {
-      if (sources[i]->getUsers().empty())
+      if (sources[i]->use_empty())
         sources[i]->erase();
       else
         break;
     }
+    // Add sinks where needed
+    auto wireType = builder.getType<quake::WireType>();
+    builder.setInsertionPoint(block.getTerminator());
+    block.walk([&](Operation *op) {
+      for (auto r : op->getResults())
+        if (r.use_empty() && r.getType() == wireType)
+          builder.create<quake::SinkOp>(builder.getUnknownLoc(), r);
```

Comment:
```suggestion
          builder.create<quake::SinkOp>(op->getLoc(), r);
```
might be better than using nothing?

Reply:
Good point. I suspect some of them will still end up with unknown locations because the injected swaps have unknown locations, but this will indeed provide valid locations for some of them.